rfc-138-enable-brotli-compression.md

```diff
@@ -0,0 +1,88 @@
+| /service-manual/service-standard   | service-manual-frontend | 254,529    | 245,881   | -3.40    |
+| /additional-commodity-code         | smart-answers           | 257,766    | 247,293   | -4.06    |
+| /government/people/theresa-may     | whitehall               | 277,888    | 265,642   | -4.41    |
+
+
+These savings for the pages tested converted into the following performance improvements. All tested on a simulated Moto G4 mobile on a 3G connection.
+
+### Homepage
+
+The visual progress of the homepage has improved by approximately 110 ms as can be seen in the visual progress graph:
+
+![Visual progress of the homepage using both compression methods.](rfc-138/visual-progress-homepage.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-homepage.png)
+
+### Coronavirus page
+
+The visual progress graph for this page starts off worse by 300ms, but quickly catches up and the viewport for Brotli completes rendering 200ms before gzip:
+
+![visual progress graph for the covid page.](rfc-138/visual-progress-covid.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-coronavirus.png)
+
+Showing a 5.53% reduction.
+
+### Bank holidays
+
+The visual progress of the bank holidays has improved by approximately 200ms as can be seen in the visual progress graph:
+
+![Visual progress of the bank holidays using both compression methods.](rfc-138/visual-progress-bank-holidays.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-bank-holidays.png)
+
+Showing a 4.60% reduction.
+
+### Search page
+
+The visual progress of the search page has improved by approximately 300ms initially, 200ms at the end. As can be seen in the visual progress graph:
+
+![Visual progress of the bank holidays using both compression methods.](rfc-138/visual-progress-search.png)
+
+And we see the a reduction in bytes for all the expected assets:
+
+![bytes for assets compared](rfc-138/total-bytes-search.png)
+
+Showing a 5.56% reduction.
+
+### Summary 
+
+In all the pages tested across the frontend apps (17 in total) we see between 3-6% reduction in the number of bytes a browser is having to download over the network. This saving is being converted to improvements in the visual metrics a users is seeing in the browser.
+
+## Proposal
+
+The proposal is to enable Brotli compression at the edge using the [Fastly Brotli Compression LA](https://www.fastly.com/release-notes/q3-2020#brotli). Browsers that support Brotli will receive these versions of the files. Any browsers that don't will receive the gzip version of the file. This can therefore be applied using the progressive enhancement methodology. In doing so we MUST update the configuration VCL so as to automate this process. During testing it was simply enabled via the Fastly user interface.
```

Comment:
In terms of how we enable this in VCL, have we done any brief investigation in how we might do this?

I found https://developer.fastly.com/reference/vcl/variables/backend-response/beresp-brotli/ but it says "Setting this will unset beresp.gzip." so does that mean we need to parse the `Accept-Encoding` header to set the right `beresp` variable accordingly?

I realise in an RFC we may not want to go too much into the implementation detail, but I just wondered if we'd already looked into it?

Reply:
Thanks for flagging @thomasleese , I've now raised a support ticket with Fastly asking for more information on how this is enabled on production and what config needs to be changed.

Reply:
In staging and integration, I just click-ops'd it in the Fastly UI.

If we wanted to do that in production, we'd need to [stop our deployment script from unsetting the compression settings](https://github.com/alphagov/govuk-cdn-config/pull/328).

Perhaps it's possible to do the same thing in pure VCL, and not touch the UI.

Alternatively, we could consider using the Fastly terraform provider to manage this, but perhaps that's a wider can of worms.

Reply:
Update from Fastly support:
> ...enabling it is the toggle and the set `beresp.brotli` in VCL since this has already been enabled for your account